paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -131,146 +347,260 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+
+      int count = is_multi_channel ? (g > 1 ? (weights_tz)[1] * (weights_tz)[0]
+                                            : (weights_tz)[0])
+                                   : 1;
+      std::vector<float> output_shift_scale(count);
+#pragma omp parallel for if (count > 1)
+      for (int i = 0; i < count; i++) {
+        if (scale_weights_data[i] == 0.0)
+          output_shift_scale[i] =
+              scale_out_data;  // weights data will contain 0
+                               // in some models, then weights
+                               // scale couldn't be calculated
+        else
+          output_shift_scale[i] =
+              scale_out_data / (scale_in_data * scale_weights_data[i]);
+      }
 
-    /* create memory descriptor for convolution without specified format
-     * ('any') which lets a primitive (convolution in this case) choose
-     * the memory format preferred for best performance
-     */
-    std::string data_format = ctx.Attr<std::string>("data_format");
-    auto chosen_memory_format =
-        platform::data_format_to_memory_format(data_format);
+      auto user_src_md = platform::MKLDNNMemDesc(
+          {src_tz}, paddle::framework::ToMKLDNNDataType(input->type()),
+          input->format());
+      auto user_weights_md = platform::MKLDNNMemDesc(
+          {weights_tz}, platform::MKLDNNGetDataType<K>(),
+          ((g) == 1) ? mkldnn::memory::format::oihw
+                     : mkldnn::memory::format::goihw);
+
+      /* create memory descriptor for convolution without specified format
+      * ('any') which lets a primitive (convolution in this case) choose
+      * the memory format preferred for best performance
+      */
+      std::string data_format = ctx.Attr<std::string>("data_format");
+      auto chosen_memory_format =
+          platform::data_format_to_memory_format(data_format);
+
+      std::vector<int> bias_tz;
+
+      auto src_md = platform::MKLDNNMemDesc(src_tz, memory::data_type::u8,
+                                            chosen_memory_format);
+      auto weights_md = platform::MKLDNNMemDesc(
+          weights_tz, memory::data_type::s8, chosen_memory_format);
+
+      auto dst_dt = fuse_relu ? paddle::framework::ToMKLDNNDataType(
+                                    framework::DataTypeTrait<uint8_t>::DataType)
+                              : paddle::framework::ToMKLDNNDataType(
+                                    framework::DataTypeTrait<int8_t>::DataType);
+
+      if (force_fp32_output) {
+        dst_dt = paddle::framework::ToMKLDNNDataType(
+            framework::DataTypeTrait<float>::DataType);
+      }
 
-    weights_format = mkldnn::memory::format::any;
-    // Check the format for user's special output
-    if (chosen_memory_format != mkldnn::memory::format::any) {
-      if (is_conv3d) {
-        chosen_memory_format =
-            platform::MKLDNNFormatForSize(src_tz.size(), chosen_memory_format);
+      float sum_scale = 1.0f;
+      if (fuse_residual_conn) {
```

Comment:
For `fuse_residual_conn`, `fuse_relu`, how about adding them in next PR? Since there is no unit-test to detect the correctness of these codes.